docs/adr/0003-End-to-end-tests.md

```diff
@@ -0,0 +1,57 @@
+# 3. End to end tests
+
+Date: 2023-04-18
+
+## Status
+
+Accepted
+
+## Context
+
+End to end (e2e) tests only run tests against exporters `/metrics` endpoint in OpenShift today (using Prow CI). We want to run them with more cases, like against Grafana, making them more reliable.
+
+## Decision
+
+e2e tests will be more user friendly.
+
+Prow CI will run e2e tests that cover more scenarios and are more reliable.
+
+### How
+
+- We will rewrite the e2e tests script in Python.
+    - easier to write and read it
+    - they will run against Pelorus operator (using test image) and not helm
+
+- We will write better documentation on what e2e tests do (script help and read the docs).
+    - more accessible to new developers of the Pelorus project
+
+- We will log only relevant information on tests run.
+    - Too much log information is not good to identify a simple error
+
+- We will not use `mig-demo-apps` anymore.
+    - make development faster (only one PR and everything in one repository)
+    - do not use yaml files for different tests scenarios
+        - use `image_name` option instead of `source_url` for faster runs (using test image)
+    - use the simplest application possible for tests (faster tests runs)
+
+- New commits/deployments/issues will be created for each test scenario.
+    - To address new approach of Pelorus metrics
+
+- We will have more tests scenarios:
+    - check each exporter `/metrics` endpoint
+    - check Grafana dashboards for each application
```

Comment:
So we will create N applications with different timestamps and so on? How would you do this with the time constrains of the timestamp() ?

Reply:
I think that is not a problem, e2e takes minutes to run, it should not hit the 30 minutes window problem (in Prow it takes sometimes more than 1 hour, but it is the setup of it, the script run takes less)

Reply:
I think this will be a problem. Some of the exporters are running against bugs or issues created in the past. Some deployments for the commits that happened in the past, so if we will add 30min window, we *will* hit this problem here.